### ISSUE TITLE
hang in uct envelope test on atomic fadd

### ISSUE THREAD

**Issue by yosefe** [openucx/ucx#350]
http://bgate.mellanox.com:8888/jenkins/job/gh-ucx-pr/794/console
```
15:20:02 [ RUN      ] rc/test_uct_perf.envelope/0
15:20:02 [     INFO ] mlx5_0:1                am latency : 0.741 usec
15:20:02 [     INFO ] mlx5_0:1                   am rate : 5.186 Mpps
15:20:02 [     INFO ] mlx5_0:1               am bcopy bw : 5027.476 MB/sec
15:20:03 [     INFO ] mlx5_0:1               am zcopy bw : 5613.386 MB/sec
15:20:03 [     INFO ] mlx5_0:1               put latency : 0.662 usec
15:20:03 [     INFO ] mlx5_0:1                  put rate : 4.421 Mpps
15:20:04 [     INFO ] mlx5_0:1              put bcopy bw : 5000.123 MB/sec
15:20:04 [     INFO ] mlx5_0:1              put zcopy bw : 7145.841 MB/sec
15:20:04 [     INFO ] mlx5_0:1               get latency : 1.274 usec
15:20:04 [     INFO ] mlx5_0:1        atomic add latency : 1.031 usec
15:20:05 [     INFO ] mlx5_0:1           atomic add rate : 2.321 Mpps
15:20:05 [     INFO ] mlx5_0:1       atomic fadd latency : 1.328 usec
15:59:31 make: *** [test] Terminated
15:59:31 Build step 'Execute shell' marked build as failure
15:59:31 TAP Reports Processing: START
15:59:31 Looking for TAP results report in workspace using pattern: **/*.tap
```

**Comment by yosefe** [comment]
can be reproduced after ~4 hours of `taskset -c 0,1 make -C build-devel/test/gtest/ test GTEST_FILTER=rc/*envelope*/* GTEST_REPEAT=1000`, iteration 207

**Comment by yosefe** [comment]
It's a race between HCA atomic operation and local store

**Comment by shamisp** [comment]
love it !

**Comment by shamisp** [comment]
fixed ?

**Comment by yosefe** [comment]
not yet. i need to rework some of the test logic.

**Comment by yosefe** [comment]
fixed by #354
